colossalai/fx/passes/algorithms/ckpt_solver_rotor.py

```diff
@@ -185,12 +186,24 @@ def _annotate_from_sequence(sequence: Sequence, node_dict: Dict[int, Node]) -> G
                 ckpt_region.append(idx)
 
 
-def solver_rotor(gm: GraphModule, data: torch.Tensor, mem_limit: int, mem_slots: int = 500) -> GraphModule:
+def solver_rotor(gm: ColoGraphModule, data: torch.Tensor, mem_limit: int, mem_slots: int = 500) -> GraphModule:
+    f"""solving the activation checkpoint in rotor's manner
+    Input:
+        gm: ColoGraphModule, we need it instead of GraphModule becuase
+        we need to generate code with activation checkpoint in colossalai's manner
+        data: the data needed for MetaInfoProp
+        mem_limit: memory limit, use Byte as unit
+        mem_slots: number of slots for discretize
+
+    Output:
+        gm: annotated graph
+        sequence: the operation sequence (for debug purpose)
+    """
     node_dict = linearize(gm)
     mem_unit = mem_limit // mem_slots
     MetaInfoProp(gm).run(data)
     chain: Chain = _construct_chain(node_dict, data, mem_unit)
     opt_table = _compute_table(chain, mem_slots)
     sequence = _rec(chain, 0, chain.length, mem_slots - chain.cweight[0], opt_table)
     _annotate_from_sequence(sequence, node_dict)
-    return gm
+    return gm, sequence
```

Comment:
Is this temporary? You may attach this sequence to GraphModule.

Reply:
sure I could set this attribute to GraphModule